.github/workflows/publish-docker.yml

```diff
@@ -0,0 +1,77 @@
+name: Publish Docker
+
+on:
+  workflow_dispatch:
+    inputs:
+      tag:
+        description: tag (ex. v0.8.3) to publish on docker
+        required: true
+
+jobs:
+  tag-docker:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        image: ["tanssi", "container-chain-simple-template", "container-chain-evm-template"]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3 
+        with:
+          fetch-depth: 0
+      - name: Login to DockerHub
+        uses: docker/login-action@v2.2.0  
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - run: |
+          DOCKER_IMAGE=moondancelabs/${{matrix.image}}
+          VERSION="${{ github.event.inputs.tag }}"
+          COMMIT=`git rev-list -n 1 '${{ github.event.inputs.tag }}'`
+          SHA=sha-${COMMIT::8}
+          echo using "${DOCKER_IMAGE}:${SHA} as base image"
+
+          mkdir -p build 
+
+          if [[ ${{matrix.image}} == 'tanssi' ]]; then
```

Comment:
This can be greatly simplified with something like:
```
      - name: Download files
        run: |
          VERSION="${{ github.event.inputs.tag }}"
          mkdir -p build
          for file in ${{ matrix.files }}
          do
            wget "${{ env.BASE_URL }}/$VERSION/$file" -O build/$file
          done
```

Reply:
apparently arrays are not as easily treatable, so I made a mix of these